src/realm/table.cpp

```diff
@@ -3340,6 +3345,17 @@ void Table::convert_column(ColKey from, ColKey to, bool throw_on_null)
             case type_Timestamp:
                 change_nullability_list<Timestamp, Timestamp>(from, to, throw_on_null);
                 break;
+            case type_ObjectId:
+                if (is_nullable(from)) {
+                    change_nullability_list<Optional<ObjectId>, ObjectId>(from, to, throw_on_null);
+                }
+                else {
+                    change_nullability_list<ObjectId, Optional<ObjectId>>(from, to, throw_on_null);
+                }
+                break;
+            case type_Decimal:
+                change_nullability_list<Decimal128, Decimal128>(from, to, throw_on_null);
+                break;
             default:
```

Comment:
Shouldn't the default be removed, so the compiler can catch new types being not covered?